commands/bnet.js

```diff
@@ -0,0 +1,443 @@
+const Discord = require('discord.js');
+
+const kCharsPerMessage = 1000;
+const kHeaderString = 'This channel stores a community collection of ' +
+'BattleTags for quick reference.\nUse ' +
+'\'[prefix] bnet [add|remove] [battleTag]\' in any channel to add or remove ' +
+'a BattleTag from your account.\n';
+
+const kSnowflakeRegex = new RegExp(/<@(\d+)>/);
+const kBattleTagRegex = new RegExp(/.{3,12}#\d{4,5}/);
+const kBattleTagOnlyRegex = new RegExp('^' + kBattleTagRegex.source + '$');
+const kBattletagLineRegex = new RegExp('(.*): (' + kBattleTagRegex.source +
+                                       ', )*('+ kBattleTagRegex.source+ ')');
+
+let channelId = null;
+let battleTags = null;
+let modRoles = null;
+
+/**
+ * A class to store a set of BattleTags for a particular user or text key. Only
+ * one of user and textId will be set. If the class is constructed with a
+ * GuildMember the user object will be stored, otherwise the specified text key
+ * will be stored.
+ * @class
+ *
+ * @property {Discord.GuildMember} user The user object this entry is for
+ * @property {string} textId The text ID this entry is for
+ * @property {string[]} battleTags A list of BattleTags to start with
+ */
+class BattleTagEntry {
+  /**
+   * Constructor for BattleTagEntry. Only one of user and text should be set.
+   * @constructor
+   *
+   * @param {Discord.GuildMember} user The user object to store
+   * @param {string} textId The text ID to store
+   * @param {string[]} battleTags A list of BattleTags to start with
+   */
+  constructor(user, textId, battleTags) {
+    this.user = user;
+    this.textId = textId;
+    this.battleTags = new Set(battleTags);
+  }
+
+  /**
+   * Adds a list of BattleTags to this BattleTagEntry. Duplicate tags will not
+   * be added (case-sensitive).
```

Comment:
does concat() remove duplicates? Comment doesn't seem in line with implementation.

Reply:
Oh wait battleTags is a Set() which is why. Maybe mention that?

Reply:
Fixed the jsdoc for the class, which is where that should be documented.